### ISSUE TITLE
Graphics artefacts

### ISSUE THREAD

**Issue by retrofan11** [BlitterStudio/amiberry#1053]
I want to report graphical artifacts in the freeware game Putty Squad (System3)(2013)(AGA).
It manifests itself in the lower part of the screen, under the publisher's logo, during loading, as well as in the main menu at the bottom of the screen, in the form of corrupted graphics.
Tried different number of settings but no success.
Another example is the demo The Castle - Amiga 64k Demo (AGA), which does not work properly, because it constantly shows double images overlapping, as well as graphical glitches immediately after starting.

Putty Squad:
1. Just put the first floppy disk into df0:
2. Click on fire after the developer logo screen to finish loading

The Castle:
1. Use HD version
2. Just click on exe file to start

OS: Debian based, DietPi 64bit (Rasberry pi3B+ and x86 PC, cpu Intel 6200)

Version 5.51
[Games.zip](https://github.com/BlitterStudio/amiberry/files/10567715/Games.zip)

**Comment by solskogen** [comment]
Is JIT on or off? Do you see the same behavior in WinUAE?

**Comment by retrofan11** [comment]
JIT is off, I use stock A1200 configuration.
It work perfect with WinUAE and same settings.

**Comment by nemo93** [comment]
@retrofan11 those are not part of the testbed. Yet I'd recommend you go first with the WHD version of Putty Squad (`PuttySquad_v1.3_AGA.lha`) and see if there's a difference.

**Comment by nemo93** [comment]
@retrofan11 those are not part of the testbed. Yet I'd recommend you go first with the WHD version of Putty Squad (`PuttySquad_v1.3_AGA.lha`) and see if there's a difference.

**Comment by retrofan11** [comment]
No, there's no difference with the whd version, it behaves exactly the same as with the original floppy disk images.
Basic A1200 configuration, fast copper off, no JIT, no turbo floppy speed...
I also tried it on the old, abandoned Uae4Arm, it works perfectly, as well as with WinUAE.
This is not some obscure game, but probably the best platformer for the Amiga.

**Comment by midwan** [comment]
@retrofan11 
Thanks for reporting this!
I can recreate the problem. It does not appear in the `dev` version, so it's a compatibility issue with the emulation.
The `dev` branch version has the latest changes from WinUAE merged, but since those make things considerably slower it's not merged in `master` yet. If you have a fast system, you can still get most games to work fine, but on an RPI3 for example, you won't even be able to get 50 FPS anymore in several. :(

**Comment by midwan** [comment]
@retrofan11 
Actually, this started to happen only recently it seems. I've tested a few previous versions of Amiberry and I didn't see the glitch there, so I'll check through the changes and see if I can fix it, by reverting what caused it.

**Comment by retrofan11** [comment]
Thank you very much for the explanation, I hope that this issue can be fixed in the latest version.
Cheers.

**Comment by midwan** [comment]
Looks like there are more than one issues here:
- The Publisher title screen glitch was caused by the memory pattern used when clearing RAM, implemented in v5.3 for the first time. I've reverted that with the commit above, and that problem is fixed.
- The game title screen still has those artifacts at the bottom, even in older versions of Amiberry (went as far back as v3.3), so it's caused by something else.

Still, the newer `dev` version does not have this, as previously mentioned. ;)
